src/config.rs

```diff
@@ -209,3 +209,31 @@ pub fn append_rpm_metadata(
 
     Ok(())
 }
+
+/// Get rpm target architecture based on the rust target
+pub fn get_target_architecture(target: &str) -> &str {
+    let mut parts = target.split('-');
+    let arch = parts.next().unwrap();
+    let abi = parts.last().unwrap_or("");
+
+    // based on the similar function from cargo-deb:
+    // https://github.com/mmstick/cargo-deb/blob/v1.23.1/src/manifest.rs#L909-L937
+    // with adjustments for valid values that `rpmbuild --target` takes
+    match (arch, abi) {
+        // https://doc.rust-lang.org/std/env/consts/constant.ARCH.html
+        // rustc --print target-list
+        // https://fedoraproject.org/wiki/Architectures
+        // https://github.com/rpm-software-management/rpm/blob/rpm-4.14.3-release/rpmrc.in#L156
+        ("mipsisa32r6", _) => "mipsr6",
+        ("mipsisa32r6el", _) => "mipsr6el",
+        ("mipsisa64r6", _) => "mips64r6",
+        ("mipsisa64r6el", _) => "mips64r6el",
+        ("powerpc", _) => "ppc",
+        ("powerpc64", _) => "ppc64",
+        ("powerpc64le", _) => "ppc64le",
+        ("x86", _) => "i386",
+        (arm, gnueabi) if arm.starts_with("arm") && gnueabi.ends_with("hf") => "armv7hl",
+        (arm, _) if arm.starts_with("arm") => "armv7l",
+        (other_arch, _) => other_arch,
+    }
+}
```

Comment:
Perhaps it'd be worth having a type that handles this mapping?

Reply:
were you thinking of a new `target` struct that would be initialized based on the rust target string, with a method to get the rpm target architecture?

I wasn't sure where to place this function actually. Maybe it would be better to make it part of the `Builder` struct implementation in builder.rs, reusing the `target` already stored there. If you prefer that too, I can go ahead and make that change.

Reply:
I was thinking of something more like `pub enum Target`, left open-ended with an `Other(String)` variant, with methods for e.g. parsing from a Rust target triple and outputting an `rpmbuild` target.

The `License` type is an example of the sort of idea in this crate:

https://github.com/iqlusioninc/cargo-rpm/blob/develop/src/license.rs

Reply:
ok got it. Will implement something similar when I got some time and post an update.

thanks for the example!

Reply:
Awesome, thank you!

Reply:
Added a commit to address this. Let me know if it looks good to you!